lib/philomena_web/views/tag_change_view.ex

```diff
@@ -15,4 +15,22 @@ defmodule PhilomenaWeb.TagChangeView do
 
   def reverts_tag_changes?(conn),
     do: can?(conn, :revert, Philomena.TagChanges.TagChange)
+
+  def tag_change_retained(%{image: image, added: true, tag: %{id: tag_id}}) do
+    Enum.any?(image.tags, &(&1.id == tag_id))
+  end
+
+  def tag_change_retained(%{image: image, added: true, tag_name_cache: tag_name}) do
+    Enum.any?(image.tags, &(&1.name == tag_name))
+  end
+
+  def tag_change_retained(%{image: image, added: false, tag: %{id: tag_id}}) do
+    not Enum.any?(image.tags, &(&1.id == tag_id))
+  end
+
+  def tag_change_retained(%{image: image, added: false, tag_name_cache: tag_name}) do
+    not Enum.any?(image.tags, &(&1.name == tag_name))
+  end
```

Comment:
by truth table,
```
x = added
y = contains
z = retained

x|y|z
0|0|1
0|1|0
1|0|0
1|1|1
```
This logic checks the same thing as `if added == contains`, so:
```suggestion
  def tag_change_retained(%{image: image, added: added, tag: %{id: tag_id}}) do
    added == Enum.any?(image.tags, &(&1.id == tag_id))
  end

  def tag_change_retained(%{image: image, added: added, tag_name_cache: tag_name}) do
    added == Enum.any?(image.tags, &(&1.name == tag_name))
  end
```